cmd/ci-operator/main.go

```diff
@@ -1285,6 +1289,47 @@ func (o *options) initializeNamespace() error {
 		return fmt.Errorf("could not update namespace to add labels, TTLs and active annotations: %w", err)
 	}
 
+	if o.restrictNetworkAccess {
+		logrus.Debugf("Creating egress firewall in namespace %s", o.namespace)
+		egressFirewall := &egressfirewallv1.EgressFirewall{
+			ObjectMeta: meta.ObjectMeta{
+				Name:      "default",
+				Namespace: o.namespace,
+			},
+			Spec: egressfirewallv1.EgressFirewallSpec{
+				Egress: []egressfirewallv1.EgressFirewallRule{
+					{
+						To: egressfirewallv1.EgressFirewallDestination{
+							NodeSelector: &meta.LabelSelector{
+								MatchLabels: map[string]string{
+									"node-role.kubernetes.io/master": "",
+								},
+							},
+						},
+						Type: "Allow",
+					},
+					{
+						To: egressfirewallv1.EgressFirewallDestination{
+							CIDRSelector: "10.0.0.0/8",
+						},
+						Type: "Deny",
+					},
+					{
+						To: egressfirewallv1.EgressFirewallDestination{
+							CIDRSelector: "0.0.0.0/0",
+						},
+						Type: "Allow",
+					},
+				},
+			},
+		}
+		if err := client.Create(ctx, egressFirewall); err != nil && !kerrors.IsAlreadyExists(err) {
+			return fmt.Errorf("could not create egress firewall: %w", err)
+		}
+	} else {
+		logrus.Debugf("Not creating egress firewall in namespace %s", o.namespace)
```

Comment:
The EgressFirewall is not being created by default. It's ok to log and inform the user when `--restrict-network-access=true` but I believe we can avoid logging when no firewall rules have been created.